File: backend/src/main/scala/microservices/attraction/objects/AttractionSuggestionListPlannerResponse.scala
package com.typesafe.travel.attraction.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class AttractionSuggestionListPlannerResponse(suggestions: List[AttractionSuggestionPlannerResponse])
object AttractionSuggestionListPlannerResponse:
  given sourceEncoder: Encoder[AttractionSuggestionListPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[AttractionSuggestionListPlannerResponse] = deriveDecoder

File: frontend/src/microservices/attraction/objects/AttractionSuggestionListPlannerResponse.ts
// 本文件定义 attraction 模块的 `AttractionSuggestionListPlannerResponse`，作为建议列表响应并提供 JSON 编解码。

import type { AttractionSuggestionPlannerResponse } from './AttractionSuggestionPlannerResponse'

export type AttractionSuggestionListPlannerResponse = {
  suggestions: AttractionSuggestionPlannerResponse[]
}

export const attractionSuggestionListPlannerResponseFromJson = (json: string): AttractionSuggestionListPlannerResponse =>
  JSON.parse(json) as AttractionSuggestionListPlannerResponse

export const attractionSuggestionListPlannerResponseToJson = (value: AttractionSuggestionListPlannerResponse): string =>
  JSON.stringify(value)
